engine/src/main/java/org/terasology/config/flexible/validators/RangedNumberValidator.java

```diff
@@ -78,4 +82,10 @@ public boolean validate(T value) {
 
         return withinMinBoundary && withinMaxBoundary;
     }
+
+    @Override
+    public void issueWarnings(T value) {
+        LOGGER.warn("Value {} is not in the range {}{}, {}{}", value, minInclusive ? "[" : "(",
+                min != null ? min : "UNBOUNDED", max != null ? max : "UNBOUNDED", maxInclusive ? "]" : ")");
```

Comment:
Your first suggestion does seem more intuitive. I'll make the code more readable.

As for your second suggestion, it will require adding a parameter to both `issueWarnings` and `validate`, either of the type `Setting` or of `SimpleUri` (for the setting's id). I prefer the former with a default value of null, so that if someone wishes to execute `validate` outside of a `Setting`, he can do so while still being able to log warnings.